cmd/cyclonexdx.go

```diff
@@ -0,0 +1,271 @@
+			Name:   kbom.GeneratedBy.Name,
+			Hashes: &[]cyclonedx.Hash{
+				{
+					Algorithm: cyclonedx.HashAlgoSHA256,
+					Value:     kbom.GeneratedBy.Commit,
+				},
+			},
+			Version: kbom.GeneratedBy.Version,
+		},
+	}
+
+	components := []cyclonedx.Component{}
+	clusterProperties := []cyclonedx.Property{
+		{
+			Name:  CdxPrefix + "k8s:component:type",
+			Value: ClusterType,
+		},
+		{
+			Name:  CdxPrefix + "k8s:component:name",
+			Value: kbom.Cluster.Name,
+		},
+		{
+			Name:  KSOCPrefix + "k8s:cluster:nodes",
+			Value: fmt.Sprintf("%d", kbom.Cluster.NodesCount),
+		},
+	}
+
+	if kbom.Cluster.Location.Location != "" && kbom.Cluster.Location.Location != "unknown" {
+		clusterProperties = append(clusterProperties, cyclonedx.Property{
+			Name:  KSOCPrefix + "k8s:cluster:location:location",
+			Value: kbom.Cluster.Location.Location,
+		})
+	}
+
+	if kbom.Cluster.Location.Region != "" {
+		clusterProperties = append(clusterProperties, cyclonedx.Property{
+			Name:  KSOCPrefix + "k8s:cluster:location:region",
+			Value: kbom.Cluster.Location.Region,
+		})
+	}
+
+	if kbom.Cluster.Location.Zone != "" {
+		clusterProperties = append(clusterProperties, cyclonedx.Property{
+			Name:  KSOCPrefix + "k8s:cluster:location:zone",
+			Value: kbom.Cluster.Location.Zone,
+		})
+	}
+
+	clusterComponent := cyclonedx.Component{
+		BOMRef:     id(kbom.Cluster),
+		Type:       cyclonedx.ComponentTypePlatform,
+		Name:       "cluster",
+		Version:    kbom.Cluster.K8sVersion,
+		Properties: &clusterProperties,
+	}
+
+	components = append(components, clusterComponent)
+
+	for i := range kbom.Cluster.Nodes {
+		n := kbom.Cluster.Nodes[i]
+		components = append(components, cyclonedx.Component{
+			BOMRef: id(n),
+			Type:   cyclonedx.ComponentTypePlatform,
+			Name:   n.Name,
+			Properties: &[]cyclonedx.Property{
+				{
+					Name:  CdxPrefix + "k8s:component:type",
+					Value: NodeType,
+				},
+				{
+					Name:  CdxPrefix + "k8s:component:name",
+					Value: n.Name,
+				},
+				{
+					Name:  KSOCPrefix + "k8s:node:osimage",
+					Value: n.OsImage,
+				},
+				{
+					Name:  KSOCPrefix + "k8s:node:arch",
+					Value: n.Architecture,
+				},
+				{
+					Name:  KSOCPrefix + "k8s:node:kernel",
+					Value: n.KernelVersion,
+				},
+				{
+					Name:  KSOCPrefix + "k8s:node:bootid",
+					Value: n.BootID,
+				},
+				{
+					Name:  KSOCPrefix + "k8s:node:type",
+					Value: n.Type,
+				},
+				{
+					Name:  KSOCPrefix + "k8s:node:operatingsystem",
+					Value: n.OperatingSystem,
+				},
+				{
+					Name:  KSOCPrefix + "k8s:node:machineid",
+					Value: n.MachineID,
+				},
+				{
+					Name:  KSOCPrefix + "k8s:node:hostname",
+					Value: n.Hostname,
+				},
+				{
+					Name:  KSOCPrefix + "k8s:node:containerruntimeversion",
+					Value: n.ContainerRuntimeVersion,
+				},
+				{
+					Name:  KSOCPrefix + "k8s:node:kubeletversion",
+					Value: n.KubeletVersion,
+				},
+				{
+					Name:  KSOCPrefix + "k8s:node:kuebproxyversion",
```

Comment:
This looks like a typo
```suggestion
					Name:  KSOCPrefix + "k8s:node:kubeproxyversion",
```

Reply:
fixed